src/main/java/io/kaoto/camelcatalog/generators/CamelCatalogSchemaEnhancer.java

```diff
@@ -321,11 +321,19 @@ private void addDefaultInfo(BaseOptionModel modelOption, ObjectNode propertyNode
         var defaultValue = modelOption.getDefaultValue();
         if (defaultValue != null && !propertyNode.has("default")) {
             var propertyType = modelOption.getType();
+            var schemaPropTypeNode = propertyNode.get("type");
+            if ("boolean".equals(schemaPropTypeNode.asText())) {
+                // some boolean properties have its type as string in the catalog. prioritize the schema if type is declared.
+                propertyType = "boolean";
+            }
+
             if ("integer".equals(propertyType) && !(defaultValue instanceof String)) {
                 propertyNode.put("default", ((BigDecimal) defaultValue).intValue());
             } else if ("boolean".equals(propertyType)) {
                 if ("true".equals(defaultValue.toString())) {
                     propertyNode.put("default", true);
+                } else if ("false".equals(defaultValue.toString())) {
```

Comment:
[Idea] We had agreed earlier to only include the default when the value is true. If it’s false, the default can be omitted since it doesn’t add any extra value. 
This is fine too, as it doesn’t change the behavior.

Reply:
OK makes sense - if I were to choose, maybe in case the default false is explicitly specified in the catalog, I'd also want to see it in the form tooltip - so let's go with this and see!